### ISSUE TITLE
Migrate the noctua-dev codebase and services to noctua; solve the slow start issue that sometimes occurs

### ISSUE THREAD

**Issue by kltm** [geneontology/noctua#183]
Migrate the noctua-dev codebase and services to noctua--the current situations is making things a little strange.

As well, the port game being played on noctua-dev on 8909 seems to be slowing down a certain set of users, or in some cases, prevent login entirely.

As a temporary workaround people should use http://noctua-dev.berkeleybop.org:8909

**Comment by kltm** [comment]
Probably another issue, but after thinking about it and how much easier it made amigo, a peripheral issue should be getting the minerva jar bundled in (which would also make unit tests much nicer to run).

**Comment by kltm** [comment]
Hm, I think I've tracked down part of the pause (I'm also going to add hang and wait for search) problem that a lot of people have been experiencing recently. Will try another experiment first and get back, but I'm thinking that this is actually a name resolution problem occuring dur to some express listen issues.
```
environment=MINERVA_DEFINITION=minerva_public,BARISTA_LOCATION="http://barista.berkeleybop.org",NOCTUA_HOST=131.243.192.110,NOCTUA_PORT=8910,MSGDEBUG=1,HOME=/home/swdev
```

**Comment by kltm** [comment]
Okay getting close. I now have it running nicely on toaster.lbl.gov:8910 (after fixing some odd things with my hosts file mapping too many things to localhost). I am still having trouble getting to to work well on noctua.berkeleybop.org, although fine on noctua.berkeleybop.org:8910 and 131.243.192.110:8910

First note that the "pause" behavior is always almost *exactly* two minutes and for a single resource--the first after the initial resource. Also, we never had this problem on OpenShift (which deployed the app with raw IP addresses, not a name).

The *only* time this seems to take place is when using proxypass in front of express when switching ports (I'm not exactly sure why barista seems immune to this, so I'll leave it for now).

My guess now is that there is an attempt somewhere at a name resolution by the browser that eventually times out and fails to the correct one. Hrm.

**Comment by kltm** [comment]
As part of the behavior, the 200 GET seems immediately successful, but it's the *receiving* part that is taking two minutes. Just had a neat one now where two resources in the page both took exactly the same two minutes two receive. Of course, getting these resources separately has not issue that I've seen yet.

**Comment by kltm** [comment]
Name looks fine, logging in and logging out looks fine to me as well. Looks good!

Final proxypass settings:
```
ProxyPass / http://toaster.lbl.gov:8910/ disablereuse=On
ProxyPassReverse / http://toaster.lbl.gov:8910/
```

Final supervisord environment:
```
environment=MINERVA_DEFINITION=minerva_public,BARISTA_LOCATION="http://barista.berkeleybop.org",NOCTUA_HOST="131.243.192.110",NOCTUA_PORT=8910,NOCTUA_LOCATION="http://noctua.berkeleybop.org",MSGDEBUG=1,HOME=/home/swdev
```

The solution went down like this. The name resolution idea wasn't going anywhere (besides a fun side trip through my hosts file, see above), but the fact that it was always exactly two minutes (120s) bugged me, so I started searching configurations in the different servers for that, then apache proxypass settings and 120 seconds; that turned up a reference to a default of ttl=120 (i.e. time to life for a connection being two minutes). Sounded promising so I read through that proxypass setting documentation and found that is does connection pooling and a bunch of other stuff that sounds like a problem waiting to happen. Turning it all off seems to have fixed it, with the culprit of the hour being reuse, which is then hanging. Fixed in future version or weird feature, who knows? Off now and things work/make sense again.

This also probably explains why barista doesn't get burned--it's usually the single call so no reuse happens and the connection closes immediately.

Anyways, noctua-dev is now down and
